be/src/exprs/vectorized/array_functions.cpp

```diff
@@ -266,36 +269,36 @@ class ArrayRemoveImpl {
     }
 
     // array is non-nullable.
-    static ColumnPtr _array_remove_non_nullable(const ArrayColumn& array, const Column& arg) {
+    static ColumnPtr _array_remove_non_nullable(const ArrayColumnPtr& array, const ColumnPtr& arg) {
         bool nullable_element = false;
         bool nullable_target = false;
         bool const_target = false;
         ColumnPtr targets_holder;
 
-        const UInt32Column& offsets = array.offsets();
-        const Column* elements = &array.elements();
-        const Column* targets = &arg;
-        if (auto nullable = dynamic_cast<const NullableColumn*>(elements); nullable != nullptr) {
+        const UInt32Column& offsets = array->offsets();
+        ColumnPtr elements = array->elements_column();
```

Comment:
This will cause the copy constructor of `std::shared_ptr` is called, which may have a little overhead.

Reply:
fixed.